test/e2e_tests/utils/testuser_adapter.go

```diff
@@ -0,0 +1,140 @@
+package utils
+
+import (
+	"context"
+	"fmt"
+
+	cappv1alpha1 "github.com/dana-team/container-app-operator/api/v1alpha1"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	ServiceAccountNameFormat = "system:serviceaccount:%s:%s"
+	ServiceAccountName       = "test-user"
+)
+
+// CreateTestUser creates a test user with the specified Kubernetes client and namespace.
+func CreateTestUser(k8sClient client.Client, namespace string) {
+	createTestUserServiceAccount(k8sClient, namespace)
+	createTestUserRoleAndRoleBinding(k8sClient, namespace)
+}
+
+// SwitchUser switches the Kubernetes client's user context to the test user if switchToTestUser is true.
+// If switchToTestUser is false, it reverts to the original context.
+func SwitchUser(k8sClient *client.Client, cfg *rest.Config, namespace string, scheme *runtime.Scheme, switchToTestUser bool) {
+	if switchToTestUser {
+		cfg.Impersonate = rest.ImpersonationConfig{
+			UserName: fmt.Sprintf(ServiceAccountNameFormat, namespace, ServiceAccountName),
+		}
+	} else {
+		cfg.Impersonate = rest.ImpersonationConfig{}
+	}
```

Comment:
Why is this `else` needed?

Reply:
The `else` block allows reverting to the original context